src/values.js

```diff
@@ -364,6 +364,52 @@ function wrapToString(type, fn) {
   }
 }
 
+/**
+ * @param {string} name A valid schema name
+ * @param {string} issuer A unique string
+ * @param {string} jwks_url A valid HTTPS URL
+ * @param {Array<Ref>} [allowed_roles=[]] A list of Role refs
+ * @param {Array<Ref>} [allowed_collections=[]] A list of user-defined Collection refs
+ *
+ * @extends module:values~Value
+ * @constructor
+ */
+function AccessProvider(
```

Comment:
Adding a specific document type for AccessProvider seems like more specificity than necessary at least compared to how the driver currently handles documents of other types. (Collection, Index, Database, Function, etc)